### ISSUE TITLE
[meta] Update dependencies

### ISSUE THREAD

**Issue by noxan** [ourownstory/neural_prophet#1306]
## Tasks

- `numpy 1.24.3` fails due to: FAILED tests/test_plotting.py::test_plot_trend[matplotlib] - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part. (see [failed github action](https://github.com/ourownstory/neural_prophet/actions/runs/4805912366/jobs/8552816843?pr=1305)).
- `torch` and `pytorch-lightning` are tracked in #1229 
- `flake8` would require `python>=3.8.1` somehow
- `holidays` returns different number of holidays from time to time
- `plotly-resampler` and `pandas` depend on another (requires `plotly-resampler=0.9.0rc0` for `pandas=2.0.1`) and fail (see [github action logs](https://github.com/ourownstory/neural_prophet/pull/1305/checks?sha=959b3719f05df59eb65cae6b632e577534b3e881))

## Outdated package list

```
flake8            5.0.4        6.0.0        the modular source code checker: pep8 pyflakes and co
holidays          0.21.13      0.23         Generate and work with holidays in Python
numpy             1.23.5       1.24.3       NumPy is the fundamental package for array computing with Python.
pandas            1.5.3        2.0.1        Powerful data structures for data analysis, time series, and statistics
pandas-stubs      1.5.3.230321 2.0.0.230412 Type annotations for pandas
pycodestyle       2.9.1        2.10.0       Python style guide checker
pyflakes          2.5.0        3.0.1        passive checker of Python programs
pytorch-lightning 1.9.5        2.0.2        PyTorch Lightning is the lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerplate.
torch             1.13.1       2.0.0        Tensors and Dynamic neural networks in Python with strong GPU acceleration
```

**Comment by leoniewgnr** [comment]
Update

- `flake8`: no update
- `holidays`: updated to newest version

**Comment by leoniewgnr** [comment]
```
bokeh                3.1.1   3.2.2       Interactive plots and applications in the browser from Python
dash                 2.9.3   2.13.0      A Python framework for building reactive web-apps. Developed by Plotly.
flake8               5.0.4   6.1.0       the modular source code checker: pep8 pyflakes and co
google-auth-oauthlib 1.0.0   1.1.0       Google Authentication Library
holidays             0.21.13 0.33        Generate and work with holidays in Python
ipython              8.12.2  8.15.0      IPython: Productive Interactive Computing
kaleido              0.2.1   0.2.1.post1 Static image export for web-based visualization libraries with zero dependencies
matplotlib           3.7.3   3.8.0       Python plotting package
numpy                1.24.4  1.26.0      Fundamental package for array computing in Python
pandas               1.5.3   2.1.1       Powerful data structures for data analysis, time series, and statistics
plotly-resampler     0.8.3.2 0.9.1       Visualizing large time series with plotly
pycodestyle          2.9.1   2.11.0      Python style guide checker
pyflakes             2.5.0   3.1.0       passive checker of Python programs
pytorch-lightning    1.9.5   2.0.9       PyTorch Lightning is the lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerp...
torch                2.0.0   2.0.1       Tensors and Dynamic neural networks in Python with strong GPU acceleration
torchmetrics         0.11.4  1.2.0       PyTorch native Metrics
```

**Comment by felixmartinsson** [comment]
Hi! How is this issue going? Wondering about upgrading pandas to 2. Would be really great. 

Thanks for great work with this package!

**Comment by ourownstory** [comment]
These should all be resolved as of 0.9.0
